botocore/endpoint.py

```diff
@@ -294,9 +294,12 @@ def create_endpoint(self, service_model, region_name=None, is_secure=True,
                 raise
         # We only support the credentialScope.region in the properties
         # bag right now, so if it's available, it will override the
-        # provided region name.
-        region_name_override = endpoint['properties'].get(
-            'credentialScope', {}).get('region')
+        # provided region name if an endpoint url was not manually set.
+        # If a endpoint url was specified, the user must specify a region.
```

Comment:
Region is only required for sigv4, so you could previously specify an endpoint-url without a region provided you weren't using sigv4.  For example, this is a breaking change for commands such as this:

```
$ aws configure list | grep region
    region                <not set>             None    None
$ aws s3api list-buckets --endpoint-url https://s3.amazonaws.com/
```